lib/_http_client.js

```diff
@@ -35,6 +34,13 @@ function ClientRequest(options, cb) {
   } else if ((agent === null || agent === undefined) &&
              typeof options.createConnection !== 'function') {
     agent = defaultAgent;
+  } else if (agent !== null && agent !== undefined &&
+            !(agent instanceof Agent.Agent)) {
```

Comment:
We should accept an instance of Agent.Agent. Had to check for the null value as well because the previous if statement also requires options.createConnection to not be a function. 

This if block could probably be rewritten slightly to account for all acceptable values and then throw the error in a simple else clause.